File: week_15/isosceles_triangle_1378.cpp
#include <iostream>
#include <math.h>

typedef std::pair<int, int> coord;

double calc_distance(coord p1, coord p2)
{
    double distance = pow((p2.first - p1.first), 2) + pow((p2.second - p1.second), 2);
    return sqrt(distance);
}

bool calc_triangle(coord p1, coord p2, coord p3)
{
    bool answer = false;

    double d_p1_p2 = calc_distance(p1, p2);
    double d_p1_p3 = calc_distance(p1, p3);
    double d_p2_p3 = calc_distance(p2, p3);

    if ((d_p1_p2 == d_p1_p3) && (d_p2_p3 != d_p1_p2))
    {
        answer = true;
    }
    else if ((d_p1_p2 == d_p2_p3) && (d_p1_p3 != d_p1_p2))
    {
        answer = true;
    }
    else if ((d_p1_p3 == d_p2_p3) && (d_p1_p2 != d_p1_p3))
    {
        answer = true;
    }

    return answer;
}

int main()
{
    while (true)
    {
        int n_coords, x, y;
        coord aux;
        std::cin >> n_coords;

        if (n_coords == 0)
            break;

        coord coords[n_coords];

        for (int i = 0; i < n_coords; i++)
        {
            std::cin >> aux.first >> aux.second;
            coords[i] = aux;
        }

        int n_sets = 0;

        for (int i = 0; i < n_coords - 2; i++)
        {
            for (int j = i + 1; j < n_coords - 1; j++)
            {
                for (int k = j + 1; k < n_coords; k++)
                {
                    if (calc_triangle(coords[i], coords[j], coords[k]) == true)
                    {
                        n_sets++;
                    }
                }
            }
        }

        std::cout << n_sets << std::endl;
    }
}

File: week_15/isosceles_triangle_1378.py
from math import sqrt


def distance(p1, p2):
    distance = (p2[0] - p1[0])**2 + (p2[1] - p1[1])**2
    return sqrt(distance)


def calc_triangle(p1, p2, p3):

    answer = False

    d_p1_p2 = distance(p1, p2)
    d_p1_p3 = distance(p1, p3)
    d_p2_p3 = distance(p2, p3)

    if d_p1_p2 == d_p1_p3 and d_p2_p3 != d_p1_p2:
        answer = True

    elif d_p1_p2 == d_p2_p3 and d_p1_p2 != d_p1_p3:
        answer = True

    elif d_p1_p3 == d_p2_p3 and d_p1_p3 != d_p1_p2:
        answer = True

    return answer


def isosceles_triangle():
    while True:
        n_coords = int(input())

        if n_coords == 0:
            break

        coords = list()
        for _ in range(n_coords):
            coords.append(tuple(map(int, input().split())))

        n_sets = 0
        for i in range(n_coords-2):
            for j in range(i+1, n_coords-1):
                for k in range(j+1, n_coords):
                    if calc_triangle(coords[i], coords[j], coords[k]):
                        n_sets += 1

        print(n_sets)


if __name__ == '__main__':
    isosceles_triangle()
